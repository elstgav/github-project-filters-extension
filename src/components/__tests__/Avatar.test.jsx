import Avatar from 'components/Avatar'
import EnzymeTestWrapper from 'test/support/EnzymeTestWrapper'

import { cachedUser } from 'test/fixtures'

const test = new EnzymeTestWrapper(Avatar)

describe('Avatar', () => {
  beforeEach(() => {
    test.props = { user: cachedUser }
  })

  it('displays a single <img>', () => {
    expect(test.Avatar.find('img')).toHaveLength(1)
    expect(test.Avatar.props().className).toBe('avatar')
    expect(test.Avatar).toMatchSnapshot()
  })

  it('has a default size', () => {
    expect(test.Avatar.props()).toMatchObject({ height: 20, width: 20 })
  })

  it('has a configurable className', () => {
    test.props.className = 'foo'
    expect(test.Avatar.props().className).toBe('avatar foo')
  })

  it('has configurable size', () => {
    test.props.size = 40
    expect(test.Avatar.props()).toMatchObject({ height: 40, width: 40 })
  })

  it('sets the alt tag to the user’s login', () => {
    expect(test.Avatar.props()).toHaveProperty('alt', `@${test.props.user.login}`)
  })

  it('sets the img src to the user’s avatar', () => {
    expect(test.Avatar.props()).toHaveProperty('src', test.props.user.avatar)
  })
})
