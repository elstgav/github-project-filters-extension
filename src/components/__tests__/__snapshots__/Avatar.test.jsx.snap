// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar displays a single <img> 1`] = `
<img
  alt="@test1"
  className="avatar"
  height={20}
  src="test1.jpg"
  width={20}
/>
`;
